WordPress/Classes/Services/SharingService.swift

```diff
@@ -452,7 +452,193 @@ public class SharingService : LocalCoreDataService
+    private func mergeSharingButtonsForBlog(blogObjectID: NSManagedObjectID, remoteSharingButtons: [RemoteSharingButton], onComplete: (() -> Void)?) {
+        managedObjectContext.performBlock {
+            var blog: Blog
+            do {
+                blog = try self.managedObjectContext.existingObjectWithID(blogObjectID) as! Blog
+            } catch let error as NSError {
+                DDLogSwift.logError("Error fetching Blog: \(error)")
+                // Because of the error we'll bail early, but we still need to call
+                // the success callback if one was passed.
+                onComplete?()
+                return
+            }
+
+            let currentSharingbuttons = self.allSharingButtonsForBlog(blog)
+
+            // Create or update based on the contents synced.
+            let buttonsToKeep = remoteSharingButtons.map { (let remoteButton) -> SharingButton in
+                return self.createOrReplaceFromRemoteSharingButton(remoteButton, blog: blog)
+            }
+
+            // Delete any cached PublicizeServices that were not synced.
+            for button in currentSharingbuttons {
+                if !buttonsToKeep.contains(button) {
+                    self.managedObjectContext.deleteObject(button)
+                }
+            }
+
+            // Save all the things.
+            ContextManager.sharedInstance().saveContext(self.managedObjectContext, withCompletionBlock: {
+                onComplete?()
+            })
+        }
+    }
+
+
+    /// Returns an array of all cached `SharingButtons` objects.
+    ///
+    /// - Parameters
+    ///     - blog: A `Blog` object
+    ///
+    /// - Returns: An array of `SharingButton`s.  The array is empty if no objects are cached.
+    ///
+    public func allSharingButtonsForBlog(blog: Blog) -> [SharingButton] {
+        let request = NSFetchRequest(entityName: SharingButton.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "blog = %@", blog)
+
+        var buttons: [SharingButton]
+        do {
+            buttons = try managedObjectContext.executeFetchRequest(request) as! [SharingButton]
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching Publicize Connections: \(error.localizedDescription)")
+            buttons = []
+        }
+
+        return buttons
+    }
+
+
+    /// Composes a new `SharingButton`, or updates an existing one, with
+    /// data represented by the passed `RemoteSharingButton`.
+    ///
+    /// - Parameters:
+    ///     - remoteButton: The remote connection representing the publicize connection.
+    ///     - blog: The `Blog` that owns or will own the button.
+    ///
+    /// - Returns: A `SharingButton`.
+    ///
+    private func createOrReplaceFromRemoteSharingButton(remoteButton: RemoteSharingButton, blog: Blog) -> SharingButton {
+        var shareButton = findSharingButtonByID(remoteButton.buttonID, blog: blog)
+        if shareButton == nil {
```

Comment:
Ahhhhhhhhhhhhhhhhh interesting. Thanks for looking up the docs, i'll definitely update few other spots.

Sorry about the noise here Eric!